### PR TITLE
Document RealmResults.{first,last} can throw

### LINKED ISSUE #1102: Document that RealmResults.first() and RealmResults.last() can throw
JavaDoc does not specify that ```RealmResults.first()``` and RealmResults.last()``` can throw an exception if no elements are found.

### PR DESCRIPTION
**Pull request by kneth**
Documenting that RealmResults.{first,last} throw exception if result is empty. I have added a simple unit test and added the exception to the documentation.

Closes #1102.

@realm/java

### CODE REVIEW

**Comment by emanuelez** [comment]
retest this please

File: realm/src/androidTest/java/io/realm/RealmResultsTest.java [hunk 1]
```diff
@@ -183,6 +183,25 @@ public void testResultListFirstIsFirst() {
         assertTrue(allTypes.getColumnString().startsWith("test data 0"));
     }
 
+    // first() and last() will throw an exception is no element exists
+    public void testResultListFirstLastThrowIfEmpty() {
```
Comment (beeender) [inline realm/src/androidTest/java/io/realm/RealmResultsTest.java]: is -> when

**Comment by beeender** [comment]
Except the the comment :+1:

**Comment by emanuelez** [comment]
:+1:

**Comment by beeender** [comment]
:+1:

**Pull request merged**